### PR TITLE
feat: simplify terms of service and email subscription

### PR DESCRIPTION
**Pull request by sandrade-dcl**
**Documentation**: https://www.notion.so/decentraland/Simplify-Terms-of-Service-and-Email-subscription-in-Explorer-d6a5780bb8574eb38a5096ecf270cd64

## How to test the changes?
1. Go to https://play.decentraland.org/?explorer-branch=feat/simplify_tos_and_email_subscription&ENABLE_new_terms_of_service_and_email_subscription
2. Enter as a new user or as a Guest.
3. In the backpack screen try 2 things:
  3.1. Click on BACK and notice you go back to the wallet screen.
  3.2. Click on NEXT and notice you go to the next step where you can enter your name and email.

image

4. Enter a name and an email (optional). Check that the [JUMP INTO DCL] button doesn't active until you type a valid name, a valid email (if you choose to enter an email) and the ToS checkbox is active.

image

5. Click on [JUMP INTO DCL] button and check the world is loaded as usual.
6. If you entered an email, also check that you have received a confirmation email in your inbox. Don't confirm it yet.

image

7. Go to your Account Settings and observe that you have a new section there called "Email notifications".
8. Enter the "Email notifications" section and check that the email that you put in the guest sign in flow is what you have there and that you have a warning below informing about that you have a confirmation awaiting in your email.

image

9. Go to your email inbox and confirm your subscription.
10. Go back to the Explorer and go out of "Email notifications" section and go back again. Observe the yellow warning has dissapear because you already confirmed your email.
11. Now, try to delete your email subscription. Simply remove the email textbox and you will receive a confirmation modal warning about the change.

image

12. If you cancel it, you will see the textbox restored with the original email.
13. If you accept it, the textbox will keep empty and that will mean that your subscription was deleted successfully.

image

14. Enter your email again and check that you receive again the confirmation email in your inbox.
15. Now, lets try to update your email. Instead of removing the textbox, simply remove the text and put another email.
16. Observe that you receive a confirmation modal warning about the change.

image

17. From this point on, you can repeat the previous steps whenever you want.

### Additional tests
Please, also test the build deactivating the `new_terms_of_service_and_email_subscription` feature flag:
1. Go to https://play.decentraland.org/?explorer-branch=feat/simplify_tos_and_email_subscription&DISABLE_new_terms_of_service_and_email_subscription
2. Check that none of the things explained above are present and everything work as currently it's working on prod.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at 00d1910</samp>

This pull request adds a sign up mode feature to the backpack editor HUD, which allows users to create and select wallets from the HUD. It also adds a feature flag to enable or disable the terms of service and email subscription consent for the sign up mode. It modifies the UI, logic, and tests of the backpack editor HUD controller and view classes to support the new feature.

### CODE REVIEW

**Review by LucasLioyQA — APPROVED** [review]
**QA reviewed and approved** by using a guest user and firefox as a browser

File: unity-renderer/Assets/DCLServices/SubscriptionsAPIService/SubscriptionsAPIService.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        UniTask DeleteSubscription(string subscriptionId, CancellationToken ct);
+        UniTask<Subscription> GetSubscription(string subscriptionId, CancellationToken ct);
+    }
+
+    public class SubscriptionsAPIService : ISubscriptionsAPIService
+    {
+        private readonly ISubscriptionsAPIClient client;
+
+        public SubscriptionsAPIService(ISubscriptionsAPIClient client) =>
+            this.client = client;
```
Comment (Kinerius) [inline unity-renderer/Assets/DCLServices/SubscriptionsAPIService/SubscriptionsAPIService.cs:19]: What's the point of wrapping up `ISubscriptionsAPIClient` with this class?
SubscriptionsAPIService implements ISubscriptionsAPIService which can be substituted by anything in the future

**Review by Kinerius — COMMENTED** [review]
Looks good!
I left one comment

File: unity-renderer/Assets/DCLServices/SubscriptionsAPIService/SubscriptionsAPIService.cs (hunk 1 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/DCLServices/SubscriptionsAPIService/SubscriptionsAPIService.cs:19]: Yep, I think you're right... I wanted to follow the same architecture that we have for the Places API and, now that I'm seeing it, it's also having the same unneeded abstraction.

If you don't mind, in order to ease the current feature's release, I'm going to annotate it and I will apply the change during the hardening week 👍

Thanks!

**Pull request merged**